builder.go

```diff
@@ -191,6 +208,23 @@ func (p buildpackPlan) has(entry BuildPlanEntry) bool {
 	return false
 }
 
+func (bom *BOMEntry) convertMetadataToVersion() {
+	if version, ok := bom.Metadata["version"]; ok {
+		bom.Version = fmt.Sprintf("%v", version)
```

Comment:
Do we want to check that `bom.Version == ""`. It seems like we shouldn't override a value that's already there.

Reply:
According to the [RFC](https://github.com/buildpacks/rfcs/blob/main/text/0043-increase-build-plan-flexibility.md#what-it-is):
```
If both a version metadata entry and the top-level version key are both set, the lifecycle should fail with a descriptive error.
```
So I thought we should have only one parameter with the value.

Reply:
@ekcasey, @natalieparellano and I agree that it doesn't really matter whether we make this change or not.
What do you think?

Reply:
Although we are checking this during detect in https://github.com/buildpacks/lifecycle/pull/369/files, I think we need to check again after build. We should check whether the destination value is set and fail if both are set and the values don't match.

Reply:
Just to make sure - you think we should keep both values, right?
I find it a bit weird to have both values set. Especially due to the quote from the RFC ([here](https://github.com/buildpacks/lifecycle/pull/367#discussion_r466647047)) but I'm ok with doing so.

Reply:
Also, when you're saying that we need to check it again after build, do you mean at the end of the `Build` function or somewhere else?

Reply:
I'm sorry, I read your comment and thought about something different.
I'll make these  changes.

Reply:
Hi @ekcasey, just want to double check:
It's clear that if both values are set and they don't match, we want to return an error, but is it ok that both values will be set to the same value? If yes, why is the quote I wrote [here](https://github.com/buildpacks/lifecycle/pull/367#discussion_r466647047 written in the RFC?

Reply:
Following our conversation, we decided that:
* If both values are set with the same value, we aren't returning an error.
* We want to delete the original value after the conversion.